### PR TITLE
chore(ci): lock mongodb version to 6.0 in tests; add e2e task against latest alpha

### PR DESCRIPTION
**Pull request by gribnoysup**
As discussed, locking the 6 MongoDB server version we are testing in tests to the non-dev 6.0

### CODE REVIEW

**Review by addaleax — COMMENTED** [review]
Just a note, but these aren’t dev versions, they are actual releases that run in Atlas, just not supported for on-prem deployment.

If we do this, we should have a follow-up ticket to re-enable them and/or test with latest-alpha as well like we do in mongosh

**Comment by gribnoysup** [comment]
I agree, I do think we should add latest alpha similar to mongosh.

For the naming, I think I am just confused by the download center layout, because the only way to find the links to these there is [behind the "development (not for production)" section](https://www.mongodb.com/download-center/community/releases/development). Thanks for clarifying

File: packages/compass-e2e-tests/helpers/compass.ts [hunk 1]
```diff
@@ -48,6 +49,31 @@ export const MONGODB_VERSION = (process.env.MONGODB_VERSION || '5.0.6')
...
+// like 999.999.999 because we are checking for `< 7.0.0-alpha` in one test)
+const LATEST_ALPHA = '6.999.999';
+
+export const serverSatisfies = (
+  semverCondition: string,
+  enterpriseExact?: boolean
+) => {
+  return (
+    semver.satisfies(
+      MONGODB_VERSION === 'latest-alpha' ? LATEST_ALPHA : MONGODB_VERSION,
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/helpers/compass.ts:66]: So this is a bit awkward while we are relying on env and not actual running version, but now that this all is in one place, we might have easier job if we decide to change this in the future?

**Comment by gribnoysup** [comment]
Okay, so adding latest alpha to e2e seems to be relative low effort, so I decided to do this in the same patch. Please tell me if you think I should split it out

**Review by addaleax — APPROVED** [review]

**Comment by gribnoysup** [comment]
Okay, so all the test failures are related to the FLE, everything else passes, so I'll merge

**Pull request merged**